crates/vim/src/motion.rs

```diff
@@ -912,6 +1004,215 @@ fn previous_word_start(
     point
 }
 
+fn previous_word_end(
```

Comment:
I have added the functions `(previous|next)_subword_(start|end)` here, but some of them were already duplicated in `crates/editor/src/movement.rs`. Should we just keep one version? It seems like the version in the vim crate covers more cases, we could move it in the editor crate and use it for both cases.

Reply:
We should re-use things where possible (and they should live in the `movement` file if they're used outside of vim).

There's a bit of a judgement call because sometimes (e.g. in #7359) it turns out the vim definition of "paragraph" is not really a useful definition outside of vim mode. I don't think that's the case here, but would need to spend more time with the diff.